src/core/utils/numbers.ts

```diff
@@ -0,0 +1,29 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { nanoid } from 'nanoid';
+import { getWindow } from './dom';
+
+/**
+ * Return input value bounded by specified range.
+ */
+export const getBoundedValue = (val: number, min: number, max: number) => {
+  return Math.min(max, Math.max(min, val));
+};
+
+/**
+ * Returns a random value using nanoid (which uses crypto API).
+ * Contains IE fallback to return a uuid using Math.random.
+ */
+export const getRandomValue = (length?: number) => {
+  if (getWindow()?.crypto) {
+    return nanoid(length);
+  }
+  /** uuid function from https://gist.github.com/jed/982883 */
+  const getUuid = (a = '') =>
```

Comment:
So maybe nanoid is not really needed anymore?

Reply:
just look at that uuid, it's very long and ugly. nanoid is shorter and faster and very lightweight

Reply:
Have you checked the inside of nanoid? 😀 Because we would still see `getRandomValue`.

Reply:
I'm not talking about the insides. The generated uuid is 36 characters long and ugly. It's just too much for our needs

Reply:
Do you really need to generate GUID? Why not some value that is 4-6 chars long? We are not working in databases where collision could happen, if we also add some prefix/suffix depending on the element, then it will be always unique.

Reply:
replaced guid and nanoid with a non-secure implementation